### ISSUE TITLE
pydantic 1.4 breaks nornir

### ISSUE THREAD

**Issue by dmfigol** [nornir-automation/nornir#473]
With release of pydantic 1.4, nornir crashes on import:
```
Traceback (most recent call last):
  File "scripts/gather_commands.py", line 4, in <module>
    from nornir import InitNornir
  File "/Users/dmfigol/projects/public/nornir-workshop/.venv/src/nornir/nornir/__init__.py", line 3, in <module>
    from nornir.init_nornir import InitNornir
  File "/Users/dmfigol/projects/public/nornir-workshop/.venv/src/nornir/nornir/init_nornir.py", line 6, in <module>
    from nornir.core.deserializer.configuration import Config
  File "/Users/dmfigol/projects/public/nornir-workshop/.venv/src/nornir/nornir/core/deserializer/configuration.py", line 146, in <module>
    class Config(BaseNornirSettings):
  File "/Users/dmfigol/projects/public/nornir-workshop/.venv/src/nornir/nornir/core/deserializer/configuration.py", line 147, in Config
    core: CoreConfig = CoreConfig()
  File "/Users/dmfigol/projects/public/nornir-workshop/.venv/lib/python3.7/site-packages/pydantic/env_settings.py", line 28, in __init__
    super().__init__(**__pydantic_self__._build_values(values, _env_file=_env_file))
TypeError: _build_values() got an unexpected keyword argument '_env_file'
```
I've had limited time to investigate this and submit a fix, but it seems the breaking change comes from this PR:
https://github.com/samuelcolvin/pydantic/pull/1011
It looks like we are using undocumented (?) API `_build_values` here: https://github.com/nornir-automation/nornir/blob/develop/nornir/core/deserializer/configuration.py#L17-L20
which now accepts an extra argument `_env_file`

**Everyone using Nornir and hit by this issue**: please pin pydantic to 1.3 in requirements.txt / pyproject.toml / Pipfile.  
If you don't use any of those, after `pip install nornir` do `pip install pydantic==1.3`

@dbarrosop @ktbyers @ogenstad 
Could one of you pin pydantic and make a release 2.3.1 (2.3 + this fix)? (flying to the conference in a couple of hours and can't fix this at the moment)

**Comment by dbarrosop** [comment]
2.3.0 uses pydantic < 1.0 so I will make a proper 2.4.0 release instead.

I think we should kill pydantic, the amount of overhead and pain it's causing it's not worth the trouble.

**Comment by ogenstad** [comment]
To be fair pydantic changed a private method that we were using, it would have been another thing if they broke the public api.

**Comment by dbarrosop** [comment]
Closed via #482
